pkg/infra/cache/redis_cache_client.go

```
@@ -0,0 +1,77 @@
+package cache
+
+import (
+	"context"
+	"crypto/tls"
+	"github.com/Azure/AzureDefender-K8S-InClusterDefense/pkg/infra/cache/wrappers"
+	"github.com/go-redis/redis/v8"
+	"time"
+)
+
+// RedisCacheClient implements ICacheClient interface
+var _ ICacheClient = (*RedisCacheClient)(nil)
+
+// RedisCacheClient redis cache client implements ICacheClient.
+// It wraps go-redis client - "github.com/go-redis/redis".
+type RedisCacheClient struct {
+	// redisClient is the redis client from "github.com/go-redis/redis".
+	// This struct delegate redisClient.
+	redisClient wrappers.IRedisBaseClientWrapper
+}
+
+// newRedisCacheClient Cto'r for RedisCacheClient
+func newRedisCacheClient(redisClient wrappers.IRedisBaseClientWrapper) *RedisCacheClient {
+	return &RedisCacheClient{
+		redisClient: redisClient,
+	}
+}
+
+// CreateRedisCacheClient is factory for RedisCacheClient
+func CreateRedisCacheClient(configuration *RedisCacheClientConfiguration) *RedisCacheClient {
+	redisClient := redis.NewClient(&redis.Options{
+		Addr:            configuration.Addr,
+		Password:        configuration.Password,
+		DB:              configuration.Db,
+		MaxRetries:      configuration.MaxRetries,
+		MinRetryBackoff: configuration.MinRetryBackoff,
+	})
+	_, _ = redisClient.Ping(context.Background()).Result()
+
```

doc

---

what exactly does it do? if this ends up when an error we should refelct it and not swallow it .. should this be a factory maybe??